lib/modules/dosomething/dosomething_organ_donation/dosomething_organ_donation.module

```diff
@@ -127,14 +127,18 @@ function dosomething_organ_donation_send_postal_code() {
  *
  */
 function dosomething_organ_donation_store_registration() {
+  // Grab this to get the sid
   $params = drupal_get_query_parameters();
 
+  // Grab this to get the uid
+  global $user;
+
   try {
-    if ($params['sid'] && $params['uid']) {
+    if ($params['sid'] && $user->uid) {
```

Comment:
it's actually on a meta tag on the page, we have grabbed it from there in the past. I think it should actually be in the Drupal.settings object...if you want to make that update.